### ISSUE TITLE
Entry links not working properly

### ISSUE THREAD

**Issue by zackkatz** [gravityview/GravityView#1021]
We've had a few reports of entry links not pointing to the proper View or Post ID.

More details to come.

**Comment by zackkatz** [comment]
https://secure.helpscout.net/conversation/578197753/14515?folderId=408730

**Comment by zackkatz** [comment]
https://secure.helpscout.net/conversation/577421340/14476?folderId=408733

**Comment by zackkatz** [comment]
This might be an issue with DataTables, and may be solved by 90bb21273e5279e2d6288919a147f16fa4de0c2b and related to #1012

**Comment by soulseekah** [comment]
This guy is not running 2.0 https://secure.helpscout.net/conversation/578197753/14515?folderId=408730 so probably unrelated.

**Comment by zackkatz** [comment]
Another here: [HS#14548](https://secure.helpscout.net/conversation/578605076/14548?folderId=408730).

I removed the "Edit Entry" internal field type and added Custom Content field with the following shortcode:

```[gv_edit_entry_link entry_id="{entry_id}" view_id="11" post_id="11" /]```

And the shortcode pointed to the root domain instead of the View.

**Comment by rafaehlers** [comment]
@zackkatz that commit above (https://github.com/gravityview/DataTables/commit/4c213be2989d48730b293efc8460f581b403453b) on DT 2.2.1 didn't work for this user: https://secure.helpscout.net/conversation/578821036/14563?folderId=408730

DT Single Entry links still pointing to the /view/ CPT.

**Comment by zackkatz** [comment]
Related to #1034 and #1030

**Comment by zackkatz** [comment]
Another DataTables report: https://secure.helpscout.net/conversation/581136624/14656/

**Comment by zackkatz** [comment]
@rafaehlers the [latest commit](https://github.com/gravityview/GravityView/commit/9b4591fdc4bb1c1c66c9b1323c11e606ecbad2da) fixes [HS#14563](https://secure.helpscout.net/conversation/578821036/14563?folderId=408735)

**Comment by zackkatz** [comment]
Still happening here [HS#14671](https://secure.helpscout.net/conversation/581720501/14671/).

- GV 2.0.9
- Multisite
- Custom Content field in a View CPT

**It appears that they're using a template override**. Perhaps the state isn't being mocked properly still. There's some funky stuff with `x_get_view`, and `X_View_Router::render`.

Logging output:

```
2018-06-05 21:17:28.766329 - ERROR --> '[error, GravityView_Entry_Link_Shortcode->get_entry] No entry defined (or entry id not valid number)'array (
  'message' => '[error, GravityView_Entry_Link_Shortcode->get_entry] No entry defined (or entry id not valid number)',
  'data' => 
  array (
    'action' => 'read',
    'view_id' => 0,
    'entry_id' => 0,
    'post_id' => 0,
    'link_atts' => '',
    'return' => 'html',
    'field_values' => '',
  ),
  'backtrace' => 'require(\'wp-blog-header.php\'), require_once(\'wp-includes/template-loader.php\'), include(\'/themes/pro-child/single-gravityview.php\'), x_get_view, X_View_Router::render, include(\'/themes/pro/framework/views/integrity/template-blank-2.php\'), x_get_view, X_View_Router::render, include(\'/themes/pro/framework/views/global/_content-the-content.php\'), the_content, apply_filters(\'the_content\'), WP_Hook->apply_filters, GV\\View::content, GV\\View_Renderer->render, GV\\View_Template->render, GV\\Gamajo_Template_Loader->get_template_part, GV\\Gamajo_Template_Loader->locate_template, load_template, require(\'/plugins/gravityview/templates/views/list.php\'), GV\\Gamajo_Template_Loader->get_template_part, GV\\Gamajo_Template_Loader->locate_template, load_template, require(\'/plugins/gravityview/templates/views/list/list-body.php\'), GV\\View_List_Template->the_field, GV\\Field_Renderer->render, GV\\Field_Template->render, GV\\Gamajo_Template_Loader->get_template_part, GV\\Gamajo_Template_Loader->locate_template, load_template, require(\'/plugins/gravityview/templates/fields/field-custom-html.php\'), GFCommon::gform_do_shortcode, do_shortcode, preg_replace_callback, do_shortcode_tag, GravityView_Entry_Link_Shortcode->read_shortcode, GravityView_Entry_Link_Shortcode->shortcode, GravityView_Entry_Link_Shortcode->get_entry, GV\\Logger->error, GV\\WP_Action_Logger->log, do_action(\'gravityview_log_error\')',
) ```
